airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source.py

```diff
@@ -40,6 +39,7 @@ def create(
         message_repository: MessageRepository,
         timeout_seconds: int = DEFAULT_TIMEOUT_SECONDS,
     ) -> "ConcurrentSource":
+        assert initial_number_of_partitions_to_generate < num_workers, "It is required to have more workeds than threads generating partitions"
```

Comment:
nit: will this raise a traced exception?

Reply:
Wouldn't it be caught by https://github.com/airbytehq/airbyte/blob/3b2dbe05ec22d3a88f434db85bfaa36dc7067db6/airbyte-cdk/python/airbyte_cdk/exception_handler.py#L34 which transform the exception as a traced one?